app/forms/apo_form.rb

```diff
@@ -7,6 +7,7 @@ class ApoForm < BaseForm
   DEFAULT_MANAGER_WORKGROUPS = %w(developer service-manager metadata-staff).freeze
 
   attr_reader :default_collection_pid
+  delegate :agreement, to: :model
```

Comment:
It looks like this form was never intended to be used in the edit use case, because the `@model` is not set until `save` is called.

Reply:
Ahh, so that explains why it was done via an accessor method like the other attributes (and the way I had it originally).  When you call edit on the apo controller (https://github.com/sul-dlss/argo/blob/master/app/controllers/apo_controller.rb#L24-L28), it passes the model into the initializer where it gets set here: https://github.com/sul-dlss/argo/blob/master/app/forms/base_form.rb#L12-L15 and then delegation works, but on the new page, the model is not sent to the initializer and you get a test failure.

Reply:
reverting back to previous version without delegation